lib/services/role_test.go

```diff
@@ -823,6 +824,94 @@ func TestCheckAccessToServer(t *testing.T) {
 				{server: serverC, login: "admin", hasAccess: false},
 			},
 		},
+		{
+			name: "one role requires MFA but MFA was not verified",
+			roles: []RoleV3{
+				{
+					Metadata: Metadata{
+						Name:      "name1",
+						Namespace: defaults.Namespace,
+					},
+					Spec: RoleSpecV3{
+						Options: RoleOptions{
+							MaxSessionTTL:     Duration(20 * time.Hour),
+							RequireSessionMFA: true,
+						},
+						Allow: RoleConditions{
+							Logins:     []string{"root"},
+							NodeLabels: Labels{"role": []string{"worker"}},
+							Namespaces: []string{Wildcard},
+						},
+					},
+				},
+				{
+					Metadata: Metadata{
+						Name:      "name1",
+						Namespace: defaults.Namespace,
+					},
+					Spec: RoleSpecV3{
+						Options: RoleOptions{
+							MaxSessionTTL: Duration(20 * time.Hour),
+						},
+						Allow: RoleConditions{
+							Logins:     []string{"root"},
+							NodeLabels: Labels{Wildcard: []string{Wildcard}},
+							Namespaces: []string{Wildcard},
+						},
+					},
+				},
+			},
+			mfaVerified: false,
+			checks: []check{
+				{server: serverA, login: "root", hasAccess: true},
```

Comment:
nit: I understand this is an existing test but I had hard time understanding the result of this test case. Perhaps it would have been easier if server variables were named after roles - e.g. server, serverWorker, serverDB, serverDB01 - it would help realize which role grants access to which server.